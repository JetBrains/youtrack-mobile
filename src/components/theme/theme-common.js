/* @flow */

import type {UIThemeColorsPaletteColors} from '../../flow/Theme';

export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: '#b8d1e5',

  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',

  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',

  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',

  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
