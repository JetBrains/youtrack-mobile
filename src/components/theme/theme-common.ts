import type {UIThemeColorsPaletteColors} from 'types/Theme';
export const themeColorsPalette: UIThemeColorsPaletteColors = {
  $greyBackground: 'rgba(184, 209, 229, 0.15)',
  $greyColor: 'rgba(59, 107, 137, 0.35)',
  $redBackground: 'rgba(219, 88, 96, 0.15)',
  $redColor: '#db5860',
  $yellowBackground: 'rgba(237, 162, 0, 0.15)',
  $yellowColor: '#eda200',
  $greenBackground: 'rgba(89, 168, 105, 0.15)',
  $greenColor: '#59a869',
  $blueBackground: 'rgba(0, 142, 255, 0.15)',
  $blueColor: '#008eff',
};
