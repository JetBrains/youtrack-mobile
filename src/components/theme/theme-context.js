/* @flow */

import React from 'react';

export const ThemeContext = React.createContext<any>(null);
