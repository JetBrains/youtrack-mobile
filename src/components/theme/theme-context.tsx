import React from 'react';
export const ThemeContext: React.Context<any> = React.createContext<any>(null);
