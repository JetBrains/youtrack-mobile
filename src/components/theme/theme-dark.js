/* @flow */

import type {UITheme} from '../../flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',

    $error: '#ee0e0e',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#b8d1e5',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)'
  }
};

export default dark;
