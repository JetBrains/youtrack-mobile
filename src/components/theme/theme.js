/* @flow */

import {Appearance} from 'react-native-appearance';
import EStyleSheet from 'react-native-extended-stylesheet';

import lightTheme from './theme-light';
import darkTheme from './theme-dark';

import type {UITheme} from '../../flow/Theme';

export const DEFAULT_THEME: UITheme = lightTheme;
export const DEFAULT_MODE: string = lightTheme.name;

export const getSystemMode = (): string => Appearance.getColorScheme() || DEFAULT_MODE;
export const DEFAULT_SYSTEM_MODE: string = getSystemMode();

export const getUITheme = (themeName: string): UITheme => {
  if ([lightTheme.name, darkTheme.name].includes(themeName)) {
    return themeName === DEFAULT_MODE ? lightTheme : darkTheme;
  }
  return DEFAULT_THEME;
};

export const buildStyles = (mode: string = DEFAULT_SYSTEM_MODE, uiTheme: UITheme = DEFAULT_THEME) => {
  EStyleSheet.build({
    $theme: mode,
    $resolved: uiTheme.colors.$icon,
    $shadowColor: uiTheme.colors.$icon,
    $link: uiTheme.colors.$link,
    $iconAccent: uiTheme.colors.$iconAccent,
    $background: uiTheme.colors.$background,
    ...uiTheme.colors
  });
};

