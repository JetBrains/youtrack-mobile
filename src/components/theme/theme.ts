import {Appearance} from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import AsyncStorage from '@react-native-async-storage/async-storage';
import darkTheme from './theme-dark';
import lightTheme from './theme-light';
import {THEME_MODE_KEY} from '../storage/storage';
import type {UITheme} from 'types/Theme';
export const DEFAULT_THEME: UITheme = lightTheme;
export const getSystemThemeMode = (): any => Appearance.getColorScheme();
export const themes: UITheme[] = [lightTheme, darkTheme];
export const getUITheme = (mode: string): UITheme => {
  const theme: UITheme | null | undefined = themes.reduce(
    (theme: UITheme | null | undefined, it: UITheme) =>
      it.mode.includes(mode) ? it : theme,
    null,
  );
  return theme || DEFAULT_THEME;
};
export const buildStyles = (mode: string, uiTheme: UITheme) => {
  EStyleSheet.build({
    $theme: mode,
    $androidSummaryFontWeight: uiTheme.androidSummaryFontWeight,
    $link: uiTheme.colors.$link,
    $resolved: uiTheme.colors.$icon,
    $disabled: uiTheme.colors.$disabled,
    $icon: uiTheme.colors.$icon,
    ...uiTheme.colors,
  });
};
export const getThemeMode = async (): Promise<string> => {
  let mode: string;

  try {
    const storedMode: string | null | undefined = JSON.parse(
      await AsyncStorage.getItem(THEME_MODE_KEY),
    );
    mode = storedMode || getSystemThemeMode();
  } catch (e) {
    mode = DEFAULT_THEME.mode;
  }

  return mode;
};
