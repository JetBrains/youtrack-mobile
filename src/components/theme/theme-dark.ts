/* @flow */

import {themeColorsPalette} from './theme-common';

import type {UITheme} from 'flow/Theme';

const dark: UITheme = {
  dark: true,
  mode: 'ytdark',
  name: 'dark',
  barStyle: 'light-content',
  androidSummaryFontWeight: '400',

  colors: {
    $background: '#1c1c1e',
    $boxBackground: '#2c2c2e',
    $dimBackground: 'rgba(0,0,0,0.7)',

    $error: '#C22731',

    $text: '#e1e1e1',
    $textSecondary: '#b0b0b0',
    $textButton: '#FFFFFF',

    $link: '#FF008C',
    $linkLight: 'rgba(254, 0, 130, 0.3)',

    $disabled: '#5d5d5d',

    $icon: '#9f9f9f',
    $iconAccent: '#577996',

    $mask: '#FFFFFF57',
    $navigation: '#696969',

    $separator: 'rgba(210, 254, 254, 0.15)',

    ...themeColorsPalette,
  },
};

export default dark;
