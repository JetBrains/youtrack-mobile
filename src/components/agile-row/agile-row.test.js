import React from 'react';

import {shallow} from 'enzyme';
import mocks from '../../../test/mocks';

import BoardRow from './agile-row';
import {buildStyles, DEFAULT_THEME} from '../theme/theme';

const cellMock = {
  id: 'id',
  row: {
    id: 'orphans',
  },
  column: {
    id: 'id',
    collapsed: false,
  },
  issues: [],
};

describe('<BoardRow/>', () => {
  let wrapper;
  let issueMock;

  beforeAll(() => buildStyles(DEFAULT_THEME.mode, DEFAULT_THEME));

  beforeEach(() => {
    issueMock = mocks.createIssueMock();
  });


  describe('Render', () => {

    beforeEach(() => {
      doShallow(createRowMock(), false);
    });

    it('should not render a row', () => {
      doShallow(null, false);
      expect(findByTestId('agileRowHeader')).toHaveLength(0);
    });

    it('should not render empty cells', () => {
      doShallow({collapsed: false}, false);
      expect(findByTestId('agileRowCells')).toHaveLength(0);
    });

    it('should not render cells if collapsed', () => {
      doShallow({collapsed: true, cells: [cellMock]}, false);
      expect(findByTestId('agileRowCells')).toHaveLength(0);
    });

    it('should render a header', () => {
      expect(findByTestId('agile-row-header')).toHaveLength(1);
    });

    it('should render a collapse button', () => {
      expect(findByTestId('agileRowCollapseButton')).toHaveLength(1);
    });

    it('should render cells', () => {
      expect(findByTestId('agileRowCells')).toHaveLength(1);
    });

    describe('Issue id', () => {
      const rowMock = createRowMock({issue: {idReadable: 'X-1'}});

      it('should not render an issue readable id', () => {
        expect(findByTestId('agileRowIssueId')).toHaveLength(0);
      });

      it('should render an issue readable id', () => {
        doShallow(rowMock);

        expect(findByTestId('agileRowIssueId')).toHaveLength(1);
      });
    });

    describe('Column', () => {
      it('should render an expanded column', () => {
        expect(findByTestId('agileRowColumn')).toHaveLength(1);
        expect(findByTestId('agileRowColumnCollapsed')).toHaveLength(0);
      });

      describe('Collapsed column', () => {
        beforeEach(() => {
          const collapsedColumnIdMock = 'columnIdCollapsed';
          const rowMock = createRowMock({
            cells: [{
              id: '',
              issues: [issueMock],
              column: {
                id: collapsedColumnIdMock,
                collapsed: true,
              },
            }],
          });
          doShallow(rowMock, true, [collapsedColumnIdMock]);
        });

        it('should render a collapsed column', () => {
          expect(findByTestId('agileRowColumn')).toHaveLength(0);
          expect(findByTestId('agileRowColumnCollapsed')).toHaveLength(1);
        });

        it('should render a color-coded rect represents a card', () => {
          expect(findByTestId('agileRowColumnCollapsedCard')).toHaveLength(1);
        });
      });
    });

    describe('Row', () => {
      it('should render a row', () => {
        doShallow(createRowMock({collapsed: false}));
        expect(findByTestId('agileRow')).toHaveLength(1);
      });

      it('should not render a collapsed row', () => {
        doShallow(createRowMock({collapsed: true}));
        expect(findByTestId('agileRowColumn')).toHaveLength(0);
      });
    });
  });


  function doShallow(row, zoomedIn: boolean = true, collapsedColumnIds: ?Array<string> = []) {
    wrapper = shallow(<BoardRow row={row} collapsedColumnIds={collapsedColumnIds} zoomedIn={zoomedIn} uiTheme={DEFAULT_THEME}/>);
  }

  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function createRowMock(...args) {
    return Object.assign({
      id: 'rowMockId',
      name: 'orphans',
      summary: 'summary',
      cells: [cellMock],
    }, ...args);
  }
});
