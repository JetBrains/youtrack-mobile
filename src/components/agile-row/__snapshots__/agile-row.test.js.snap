// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardRow/> Render should render a snapshot 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID="agileRow"
>
  <View
    style={
      Object {
        "paddingBottom": 16,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 16,
      }
    }
    testID="agileRowHeader"
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
      testID="agileRowCollapseButton"
    >
      <IconAngleDownRight
        color="#1f2326"
        isDown={true}
        size={20}
        style={
          Object {
            "marginTop": 2,
            "width": 16,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#444444",
              "fontSize": 20,
              "fontWeight": "500",
              "letterSpacing": 0.13,
              "lineHeight": 24,
              "marginLeft": 8,
            },
            Object {
              "fontSize": 16,
            },
            undefined,
          ]
        }
      >
        orphans
      </Text>
    </TouchableOpacity>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 0.5,
        "borderColor": "#CCC",
        "flexDirection": "row",
      }
    }
  >
    <AgileRowColumn
      cell={
        Object {
          "column": Object {
            "collapsed": false,
            "id": "id",
          },
          "id": "id",
          "issues": Array [],
          "row": Object {
            "id": "orphans",
          },
        }
      }
      key="id"
      lastColumn={true}
      testID="agileRowColumn"
    />
  </View>
</View>
`;
